assets/content/cookbook/Other/search-for-files-with-extension.md

```diff
@@ -0,0 +1,38 @@
+# Search for all files of particular extension
+
+This example uses Haxe's Filesystem to recursively search for all files of certain extension. 
+
+```hx
+static function recursiveSearch(directory:String = "path/to/", extension:String) {
+    var files:Array<String> = [];
+    
+    if (sys.FileSystem.exists(directory)) {
+      for (file in sys.FileSystem.readDirectory(directory)) {
+        var path = haxe.io.Path.join([directory, file]);
+        if (!sys.FileSystem.isDirectory(path)) {
+        // regex expression to search for files with a '.'
+        var re = ~/[^.]+$/;
+        if (re.match(path)) {
```

Comment:
Since `haxe.io.Path` is already used earlier, maybe `Path.extension()` should be used here instead?

Reply:
@Gama11 I have added an import for `haxe.io.Path`

Thanks

Reply:
The import isn't what I meant (though I agree it's nicer with it). I was wondering if we should use `Path.extension()` rather than a regex to detect the file extension. Or does the regex work better for some reason?

Reply:
@Gama11 yes I later figured.